tools/gohan_client_bash_completion.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+
+commands="list show create set delete"
+verbosity="0 1 2"
+outputFormat="json table"
+gohanCommandPos=1
+getUnnamedProperties()
+{
+device=$1
+OIFS=$IFS
+IFS=$'\n' arr=`gohan client $device list`
+wasProperties=0
+counter=0
+unnamed=""
+for a in $arr
+do
+        cop=$a
+        replacewhat="-"
+        replacewith=""
+        replacewhat2="+"
+        result="${cop//$replacewhat/$replacewith}"
+        result="${result//$replacewhat2/$replacewith}"
+        let counter=counter+1
+        if [  -z $result ] ; then
+        continue
+        fi
+        if [[ $counter -gt 2 ]];then
+        args=$(echo $a | tr "|" "\n")
+        sub=0
+        for i in $args
+        do
+                let sub=sub+1
+                if [ $sub -eq 1 ] && [ ! -z "${i// }" ] ; then
+			i="$(echo -e "${i}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
+                        unnamed="$i $unnamed"	
+                fi
+        done
+        fi
+done
+}
+
+getProperties()
+{
+named="--output-format --verbosity --fields"
+fields=""
+device=$1
+OIFS=$IFS
+IFS=$'\n' arr=`gohan client $device`
+wasProperties=0
+for a in $arr
+do
+        if [[ wasProperties -eq 1 ]] ; then
+        curvar=`echo $a | awk '{print $2}'`
+        if [[ -z "${curvar// }" ]] ; then
+        return
+        else
+	curvar="$(echo -e "${curvar}" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')"
+        named="$named --$curvar"
+	fields="$fields $curvar"
+        fi
+        fi
+
+        if [[ $a == *"Properties:"* ]] ; then
+        wasProperties=1
+        elif [[ -z "${a// }" ]] ; then
+        wasProperties=0
+        fi
+done
+
+}
+
+getAllSchemas()
+{
+OIFS=$IFS
+IFS=$'\n'
+array=`gohan client`
```

Comment:
This doesn't work when the server can't be reached or when a required environment variable (like `GOHAN_SERVICE_NAME`) is not set.

Checking `gohan client`'s return value would fix it.